### DISCUSSION TITLE
get_vertices() in Gridgen class only returns vertices in 2D plane

### DISCUSSION THREAD

**Discussion by flydream0428** [modflowpy/flopy#1974] [Q&A]
This may be purposely designed. But I expect to see 8 vertices rather than 4 when I call this function. Is there a good reason for this? Is there another function to get the 3D vertices?

**Comment by wpbonelli** [comment]
`Gridgen.get_vertices()` was likely designed/written for use mainly by `Gridgen` internally. `Grid` classes have an `xyzvertices` property (3-tuple of x, y and z coordinate lists) that can be used to process 3D geometry once you've built the grid. Apologies for the late reply here.

**Comment by christianlangevin** [comment]
Because the cells in MODFLOW have flat tops and bottoms, there is not much to be gained by having all 8 vertices.  It is a bit more efficient to access the 2D vertices and the cell top and bottom elevations, which should be available through Gridgen.
